sources/api/netdog/src/cli/update_responder.rs

```diff
@@ -0,0 +1,130 @@
+#[cfg(net_backend = "systemd-networkd")]
+use super::{error, primary_interface_name, Result};
+#[cfg(net_backend = "systemd-networkd")]
+use crate::dns::DnsSettings;
+#[cfg(net_backend = "systemd-networkd")]
+use crate::networkd_status::{get_link_status, NetworkdStatus};
+#[cfg(net_backend = "systemd-networkd")]
+use crate::{CURRENT_IP, PRIMARY_SYSCTL_CONF, SYSCTL_MARKER_FILE, SYSTEMD_SYSCTL};
+#[cfg(net_backend = "systemd-networkd")]
+use argh::FromArgs;
+#[cfg(net_backend = "systemd-networkd")]
+use snafu::{ensure, ResultExt};
+#[cfg(net_backend = "systemd-networkd")]
+use std::fmt::Write;
+#[cfg(net_backend = "systemd-networkd")]
+use std::fs;
+#[cfg(net_backend = "systemd-networkd")]
+use std::net::IpAddr;
+#[cfg(net_backend = "systemd-networkd")]
+use std::path::Path;
+#[cfg(net_backend = "systemd-networkd")]
+use std::process::Command;
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "update-responder")]
+#[cfg(net_backend = "systemd-networkd")]
+/// Updates the various files needed when responding to events
+pub(crate) struct UpdateResponderArgs {}
+
+/// Updates the various files needed when responding to events
+#[cfg(net_backend = "systemd-networkd")]
+pub(crate) fn run() -> Result<()> {
+    let primary_interface = primary_interface_name()?;
```

Comment:
The name should have settled because this will be called once the interface is `configured` so the name shouldn't change after that. This comes pretty late in the process you described.